### PR TITLE
Add custom Microsoft.Extensions.Configuration provider for Pkl modules in ASP.NET Core

### PR DESCRIPTION
**Pull request by Rafaeruo**
This adds a custom Microsoft.Extensions.Configuration provider for Pkl modules as well as an example ASP.NET Core project.

### CODE REVIEW

**Comment by jeremyvaartjes** [comment]
@Rafaeruo is the `Microsoft.Extensions.Configuration` name going to cause complexities with the fact that if we ever want to publish the nuget package, we will run into a prefix reservation problem. https://learn.microsoft.com/en-au/nuget/nuget-org/id-prefix-reservation

File: Pkl/StdOutput/MessagePackStandardOutputReader.cs [hunk 1]
```diff
@@ -23,22 +24,27 @@ public void Start()
     }
 
     private void BeginRead()
-    {
+    {
```
Comment (jeremyvaartjes) [inline Pkl/StdOutput/MessagePackStandardOutputReader.cs]: Looks like we got some extraneous spaces here, but more of a nit pick than something we need to fix.

**Comment by Rafaeruo** [comment]
> @Rafaeruo is the `Microsoft.Extensions.Configuration` name going to cause complexities with the fact that if we ever want to publish the nuget package, we will run into a prefix reservation problem. https://learn.microsoft.com/en-au/nuget/nuget-org/id-prefix-reservation

Yeah I think you're right, the `Microsoft.Extensions.Configuration` prefix really is reserved.
image

We should probably rename it, I think I'll go with `Pkl.Microsoft.Extensions.Configuration` for now. If you think we should follow a different naming convention let me know.

As for the namespace, I'm leaning towards keeping it `Microsoft.Extensions.Configuration.Pkl`, any thoughts @jeremyvaartjes?

**Comment by jeremyvaartjes** [comment]
Happy to keep the namespace the same for now, easy to change later anyway.

As for the package ID, might be worth considering something more like PklCsharp or any other alternative you have in mind, just because if we want to reserve our own prefix, we will hit this rule with "pkl" only in the name

> Avoid ID prefix reservations that are shorter than four characters and avoid common or generic words.

**Pull request merged**